app/uk/gov/hmrc/gform/formtemplate/JsonSchemeValidator.scala

```diff
@@ -32,20 +34,67 @@ object JsonSchemeValidator {
+    "multivalue"       -> List(TypeChoiceOrRevealingChoice),
+    "hints"            -> List(TypeChoiceOrRevealingChoice),
+    "optionHelpText"   -> List(TypeChoiceOrRevealingChoice),
+    "dividerPosition"  -> List(TypeChoiceOrRevealingChoice),
+    "noneChoice"       -> List(TypeChoiceOrRevealingChoice),
+    "noneChoiceError"  -> List(TypeChoiceOrRevealingChoice),
+    "dividerText"      -> List(TypeChoiceOrRevealingChoice),
+    "displayCharCount" -> List(TypeText, MultiLineTrue),
+    "dataThreshold"    -> List(TypeText, MultiLineTrue)
+  )
+
   def checkSchema(json: String): Either[SchemaValidationException, Unit] = parser.parse(json) match {
     case Right(json)          => validateJson(json)
     case Left(parsingFailure) => Left(SchemaValidationException("Json error: " + parsingFailure))
   }
 
-  private def validateJson(json: Json): Either[SchemaValidationException, Unit] =
+  def validateJson(json: Json): Either[SchemaValidationException, Unit] =
     parsedSchema match {
       case Left(parsingFailure) => Left(SchemaValidationException("Schema error: " + parsingFailure))
       case Right(schema) =>
         val formTemplateSchema: Schema = Schema.load(schema)
         val validated: ValidatedNel[ValidationError, Unit] = formTemplateSchema.validate(json)
-        validated
-          .leftMap(errors => SchemaValidationException(errors.map(_.getMessage)))
-          .toEither
+
+        validated.leftMap { errors =>
+          val conditionalValidationErrorMessages: List[String] = {
+            val indexOfAnyOf = errors.map(_.keyword).toList.indexOf("anyOf")
+
+            if (indexOfAnyOf > 0) {
+              val allConditionalValidationErrors: List[(String, String)] = errors.toList
+                .slice(indexOfAnyOf, errors.length)
+                .flatMap { error =>
+                  error.keyword match {
+                    case "not" =>
+                      val errorField: String = "\\[\".+\"]".r.findAllIn(error.getMessage).next()
+                      Some((errorField.slice(2, errorField.length - 2), error.location))
+                    case _ => None
+                  }
+                }
+
+              val deduplicatedConditionalValidationErrors = ListSet(allConditionalValidationErrors).flatten
+
+              val formattedConditionalValidationErrors: List[String] =
+                deduplicatedConditionalValidationErrors.map { case (errorProperty, errorLocation) =>
+                  s"$errorLocation: Property $errorProperty can only be used with ${conditionalRequirements(errorProperty)
+                    .mkString(", ")}"
+                }.toList
+
+              errors.map(_.getMessage).toList.slice(0, indexOfAnyOf) ++ formattedConditionalValidationErrors
+            } else {
+              errors.map(_.getMessage).toList
+            }
+          }
+
+          SchemaValidationException(
+            NonEmptyList(conditionalValidationErrorMessages.head, conditionalValidationErrorMessages.tail)
```

Comment:
is that possible that `conditionalValidationErrorMessages is empty?
Also you can use NonEmptyList.fromList method instead here or its unsafe version.

Reply:
also `errors` on line 65 is NonEmptyList and the final result is NonEmptyList. May be we should keep it as `Nel` and never convert it to list, to ensure non-emptiness?

Reply:
oops, just saw @josef-vlach comments about the same...